lib/config.js

```diff
@@ -215,15 +215,29 @@ function createEnvironmentConfig(envs, reset) {
  * @param {string} [cwd] current working directory. Defaults to process.cwd()
  */
 function Config(options) {
-    var useConfig;
+    var useBaseConfig,
+        useConfig;
+
     options = options || {};
 
     this.ignore = options.ignore;
     this.ignorePath = options.ignorePath;
     this.cache = {};
 
-    this.baseConfig = options.reset ? { rules: {} } :
-            require(path.resolve(__dirname, "..", "conf", "eslint.json"));
+    useBaseConfig = options.baseConfig;
```

Comment:
Please add some comments here explaining this behavior - it's not very obvious just from looking at it.